### PR TITLE
fix(content): Add minor tag and extra formatting to Repulsor Failure

### PR DESCRIPTION
**Pull request by Terin**
This PR cleans up formatting for the Repulsor Failure mission, and adds a minor tag. Without it, you can have an instance of First Contact with aliens, followed by text saying that you're somehow still in space and haven't fully entered the planet's atmosphere yet. This will ensure that it won't collide with any landing missions.

### CODE REVIEW

**Comment by Terin** [comment]
Has #5930 been resolved since that was merged, where minor and landing now interact properly? Since that appears to be the case here. Having the minor tag prevents Repulsor Failure from appearing when there's another (landing) mission. Without it, you can get collisions. Should that error message be removed now?

**Comment by Hurleveur** [comment]
No idea about your questions but whilst I think of it: shouldn't you make this mission invisible? Because rn when you take off you see this as having the mission to land again. 
It could also fail itself once you launch after showing the dialog? Idk how well that would work
Also you get fuel and all as if you had landed but we can't do much about that I think

**Comment by Terin** [comment]
Yep, was just retesting it and noticed the marker on the map when taking a screenshot, will add that.

**Comment by Anarchist2** [comment]
How does the mission interact when completing another job?

**Comment by Zitchas** [comment]
I'm not sure we have the mechanics for it, but ideally this would either
a) be forcibly blocked by *any* part of a mission triggering when we land on the planet. (I think minor just blocks it from happening when other missions are *starting* here, but I'm not sure)
b) forcibly block *any* part of a mission triggering when we land on the planet. (at least, until after we've successfully landed, at which point stuff should trigger normally.)

I can see use cases for both of these, but given that this is primarily intended as random filler to make some otherwise uninteresting landing interesting, I'd suggest option A. Yes, technically it could happen anywhere, but from a story perspective it's more helpful to make a boring landing into an interesting one, than to add a bit more un-accounted-for drama to an already existing mission. (given that at least a few missions describe landing, this might cause problems)

**Comment by Amazinite** [comment]
Here's the issue on the matter: #4752
And here's the relevant code for how the game chooses which missions to offer:
https://github.com/endless-sky/endless-sky/blob/97f9c300d6d30ae9f04d47667239de49c75b7f38/source/PlayerInfo.cpp#L2694-L2722

As can be seen, if hasPriorityMissions is true, *only* those missions which are offered at the spaceport and don't have priority get removed. This means that `landing` and `priority` definitely don't mix, and as can be seen from #4752, I described that as the case. That should mean though that all we need to get `landing` and `priority` working together is to remove the location check.

But when the game is going through checking for `minor` missions, there's no such location check. The game simply whittles the available missions list down to one mission, assuming all available missions are minor, otherwise all minor missions get removed and only the non-minor, non-priority missions remain. So `minor` should definitely work with `landing`.

Not sure why I thought that `minor` and `landing` don't mix, as if you check the above code on the commit that I linked in the issue there have been no changes since then; perhaps I did some testing with `minor` and `landing` and they didn't appear to work for some reason, although it's been two years since that issue and I only described how to replicate `priority` and `landing` not mixing so maybe my `minor` and `landing` test was faulty.

**Comment by Amazinite** [comment]
> I'm not sure we have the mechanics for it, but ideally this would either
a) be forcibly blocked by any part of a mission triggering when we land on the planet. (I think minor just blocks it from happening when other missions are starting here, but I'm not sure)
b) forcibly block any part of a mission triggering when we land on the planet. (at least, until after we've successfully landed, at which point stuff should trigger normally.)

So `minor`'s behavior is to not offer if something else can offer and `priority`'s behavior is to not allow other non-priority mission to offer. I suppose what you're looking for is some keyword that gives an even stronger guarantee of when it offers; a `major` keyword where only one major mission can offer at once, it only offers if no other missions are active, and it blocks all other missions from offering so long as it is active? (Having a keyword that ensures that the mission is the sole mission active seems simpler to me than trying to block other missions from running their actions if this special mission runs an action.)

**Comment by Terin** [comment]
Yeah, that'd be about what we'd need to make this work. Although the player would almost certainly have some mission active, like the Archon ones. I guess we could just make it non-invisible missions? Ideally, it would just trigger if none of your missions have any interaction with the planet you're landing on, but not sure how simple of a check that would be.

Another issue that was brought up on the Discord is that you can accept new missions from the job board via the Map > Missions tab. Although this could also apply to any mission that has you depart/flee/die on landing, like the Navy inspection in early Free Worlds, so it's either a broader issue we address elsewhere, or just overlook since the player kinda has to go out of their way to do.

>Not sure why I thought that minor and landing don't mix, as if you check the above code on the commit that I linked in the issue there have been no changes since then; perhaps I did some testing with minor and landing and they didn't appear to work for some reason, although it's been two years since that issue and I only described how to replicate priority and landing not mixing so maybe my minor and landing test was faulty.

Seems like we'd be safe to tweak the test to just check for priority and landing, then? I'll do that here so we can at least get it to build, and I'll poke around a bit more to see if there are any other issues.

**Comment by tehhowch** [comment]
It kinda seems like this whole mission should be removed; it says the player didn't land but a whole day goes by, and all the other things associated with that.

This mission instead would be suited for a "prelanding" offering location, which would be active while your ship is displayed as descending to the surface.

**Comment by petervdmeer** [comment]
> It kinda seems like this whole mission should be removed; it says the player didn't land but a whole day goes by, and all the other things associated with that.

The idea that the player loses a day to perform the repairs in-orbit when re-orbiting is actually part of the mission, but the ability to finish other missions and having the ship refuelled and repaired while re-orbiting is indeed not intended.

Terins fixes in this PR are good and do solve _some_ of the major issues that the Repulsor failure mission has. If we want to remove the repulsor failure mission, then I suggest to merge Terins fixes first and then create a separate PR/commit with only the removal of the mission. Such a separate remove PR/commit will allow the "revert of the remove" to be done as a complete set once the missing features are in the game codebase.

> This mission instead would be suited for a "prelanding" offering location, which would be active while your ship is displayed as descending to the surface.

Yes, ticket #6225 seems to request exactly this feature.

**Review by petervdmeer — APPROVED** [review]

**Review by Amazinite — CHANGES_REQUESTED** [review]
I'm gonna use this PR as a chance to do a bit of review on a mission I didn't review before it was merged. (Smh I would have caught the indents.)
* The source shouldn't need four `not attributes` lines; you should be able to just have `not attributes "station" "gas giant" "uninhabited" "korath"`.
  * A `not` on a location filter is saying "If the location matches this filter, then don't offer the mission." All attributes on a single line means that the location matches if it has the first attribute or the second attribute or the third attribute and so on. So if a planet matches "station" or "gas giant" or "uninhabited" or "korath", then it trips the not filter. A not filter on all attribute on a single line is not the same thing as applying a not boolean operation to a list of things that are or'd together. That is, it's not flipping it to "not 'station' and not 'gas giant' and not ..." which would fail if the planet had one of those attributes. I mention that because I imagine that's the reason why the `not attributes` were all on separate lines.
* The random chance is way too high. With this being able to offer on any landing, a random of < 8 means you're more likely than not to see this within your first 9 landings after hitting the 2 mil in spare credits mark.
  * A random of < 1 makes more sense as that would take about 70 landings before you're more likely than not to have seen it, but given that this can occur on any landing I wouldn't be opposed to going even lower and having something like `random < 1` and then on a separate line `random < 50`, meaning it'd take about 140 landings to hit that 50/50 point.

**Comment by Amazinite** [comment]
>Not sure why I thought that `minor` and `landing` don't mix.

I think I figured it out: `minor` only blocks multiple missions from offering at the same time, but I believe I was testing Face to Maw after 62f0226, and the situation there was that one mission was completing while Face to Maw was offering, so Face to Maw doesn't get blocked because of `minor`.

**Pull request merged**
